inference.py

```diff
@@ -43,112 +46,214 @@
-                                                                 aug_type='totensor')
-            sample['sat_img'] = img_array[row:row + chunk_size, col:col + chunk_size, :]
-            logging.debug(f"Sample shape: {sample['sat_img'].shape}")
-            sample = totensor_transform(sample)
-            inputs = sample['sat_img'].unsqueeze_(0)
-            inputs = inputs.to(device)
-            output_lst = []
-            for transformer in transforms:
-                # augment inputs
-                augmented_input = transformer.augment_image(inputs)
-                augmented_output = model(augmented_input)
-                if isinstance(augmented_output, OrderedDict) and 'out' in augmented_output.keys():
-                    augmented_output = augmented_output['out']
-                logging.debug(f'Shape of augmented output: {augmented_output.shape}')
-                # reverse augmentation for outputs
-                deaugmented_output = transformer.deaugment_mask(augmented_output)
-                deaugmented_output = F.softmax(deaugmented_output, dim=1).squeeze(dim=0)
-                output_lst.append(deaugmented_output)
-            outputs = torch.stack(output_lst)
-            outputs = torch.mul(outputs, WINDOW_SPLINE_2D)
-            outputs, _ = torch.max(outputs, dim=0)
-            outputs = outputs.permute(1, 2, 0).argmax(dim=-1)
-            outputs = outputs.reshape(chunk_size, chunk_size).cpu().numpy()
-            if debug:
-                logging.debug(f'Bin count of final output: {np.unique(outputs, return_counts=True)}')
-            pred_img[row:row + chunk_size, col:col + chunk_size] = outputs
-    pred_img = pred_img[padding:-padding, padding:-padding]
+    padded = chunk_size * 2
+    h = input_image.height
+    w = input_image.width
+    h_ = h + padded
+    w_ = w + padded
+    fp = np.memmap(tp_mem, dtype='float16', mode='w+', shape=(h_, w_, num_classes))
+    dist_samples = int(round(chunk_size * (1 - 1.0 / 2.0)))
+
+    # construct window for smoothing
+    WINDOW_SPLINE_2D = _window_2D(window_size=padded, power=2.0)
+    WINDOW_SPLINE_2D = torch.as_tensor(np.moveaxis(WINDOW_SPLINE_2D, 2, 0), ).type(torch.float)
+    WINDOW_SPLINE_2D = WINDOW_SPLINE_2D.to(device)
+
+    sample = {'sat_img': None, 'map_img': None, 'metadata': None}
+    cnt = 0
+    img_gen = gen_img_samples(input_image, chunk_size)
+    for img in tqdm(img_gen, position=1, leave=False, desc='inferring on window slices'):
+        row = img[1]
+        col = img[2]
+        sub_image = img[0]
+        image_metadata = add_metadata_from_raster_to_sample(sat_img_arr=sub_image,
+                                                            raster_handle=input_image,
+                                                            meta_map={},
+                                                            raster_info={})
+
+        sample['metadata'] = image_metadata
+        totensor_transform = augmentation.compose_transforms(params,
```

Comment:
`params` is called but never asign

Reply:
I investigated this behavior; `params` is not passed to the local function but called globally. Difficult to detect because no error is thrown. This is not ideal and has been fixed with the latest commit. Thanks for pointing it out.